test/common/ops/changeClass.js

```diff
@@ -27,6 +28,19 @@ describe('shared.ops.changeClass', () => {
     }
   });
 
+  it('req.query.class is an invalid class', (done) => {
+    user.flags.classSelected = false;
+    user.preferences.disableClasses = false;
+
+    try {
+      changeClass(user, {query: {class: 'cellist'}});
+    } catch (err) {
+      expect(err).to.be.an.instanceof(BadRequest);
+      expect(err.message).to.equal(i18n.t('invalidClass'));
+      done();
+    }
```

Comment:
Oh thanks for the reply! I didn't know that's what `done()` did. That makes sense.

Since that's apparently what the other tests do, it's probably fine. However, it seems sub-optimal to sit and wait for a timeout for every possible failure case. But it might not be a big deal. Can someone else clarify please? @staff